crawl-ref/source/l-view.cc

```diff
@@ -199,6 +200,27 @@ LUAFN(view_cell_see_cell)
     return 1;
 }
 
+/*** Is there a timed portal on this level?
+ * @treturn boolean
+ * @function timed_portal
+ */
+LUAFN(timed_portal)
+{
+    for (rectangle_iterator ri(BOUNDARY_BORDER - 1); ri; ++ri)
+    {
+        const coord_def p = *ri;
+        dungeon_feature_type feat = env.grid(p);
+        if (feat_is_portal_entrance(feat) && feat != DNGN_ENTER_TROVE && feat != DNGN_ENTER_ZIGGURAT)
```

Comment:
Do we also need to check if there's an untimed bazaar? Would it be better to somehow look for the lua logic that runs these timers, rather than scanning for features?

Reply:
Finding the list of timers would definitely be a better approach, but I looked through the dlua code that controls markers and timed messages, and I couldn't make any sense of it.  And there's nothing that indicates whether a branch or entrance is timed or not.

Unless there's someone who understands the timed message system well enough -- who could offer a suggestion as to how to figure out if there's a timer on the level -- maybe it's best to just close this PR.  Keeping the list of timed portals up-to-date by manually updating this function whenever portals change would be a pain to manage.

Reply:
Closing seems fine for now. We can re-open if we find a better idea for how to manage this.